app/views/shared/_dislike.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= link_to(like_path(post_id: @post, like_id: @post.likes.find_by(user_id: current_user.id).id), id: "dislike_#{@post.likes.find_by(user_id: current_user.id).id}", method: :delete, data: {remote: true}, class: "dislike") do%>
+    <i class="material-icons">favorite</i>
+<%end%>
+
+<div id = "post-link_<%= @post.id%>"class = "post-link">
+    <%= link_to  post_path(@post) do %>
+        <i class="material-icons">forum</i>
+    <%end%>
+</div>
+<div id="like-counter_<%= @post.id%>" class="likes-number">
+    <h2><%= @post.likes.count > 1 ? "#{@post.likes.count} Likes" : "#{@post.likes.count} Like"%></h2>
```

Comment:
- [ ] **[OPTIONAL]** `@post.likes.count` will perform queries to the database as it will count the number of likes columns for every post. As your application grows this will slow down your application so you need another mechanism to count the likes, here comes the `counter_cache` which basically cache the `count` and save it into separate column in `posts` table.
[1](https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html) [2](https://stackoverflow.com/questions/56913688/how-to-get-the-size-of-belong-to-record/56916353#56916353)